src/__tests__/useAsync.test.tsx

```diff
@@ -0,0 +1,131 @@
+import { useCallback } from 'react';
+import { cleanup, renderHook } from 'react-hooks-testing-library';
+import useAsync from '../useAsync';
+
+afterEach(cleanup);
+
+// NOTE: these tests cause console errors.
```

Comment:
idk what to do about this. :man_shrugging: 
never used react-hooks-testing-library before, and I'm generally not a fan of _not_ testing UI -targeting code outside the browser :-\

Reply:
I think it is pretty standard to test React components or hooks using Jest.

Reply:
I'm challenging that standard ;)

Reply:
it'd be a pretty big change, but https://www.bigtestjs.io/ (a layer on top of mocha)
It's more for UI testing, but mocha also does unit testing.

The main thing is that we'd be able to set up real react scenarios and test interaction with real components in a real browser in a real dom. 

(users don't use fake dom, is my main point here)

Reply:
alternatively, if the react team had more ergonomic testing tools for hooks that considered the async / effect case, I think that would still get us a long way

Reply:
my main thing with the current react-hooks-testing-library are the console errors, and how the API for getting the next value is kinda awkward -- requires a `.rerender(..props)` and `waitForNextUpdate` :-\